modules/core/src/test/scala/org/scalasteward/core/edit/RewriteTest.scala

```diff
@@ -931,6 +932,37 @@ class RewriteTest extends FunSuite {
     runApplyUpdate(update, original, expected)
   }
 
+  test("duplicate updates should be successful") {
+    val dependency = "com.pauldijou".g % "jwt-play-json".a % "5.0.0"
+
+    val artifactId = Update.ForArtifactId(
+      CrossDependency(dependency),
+      newerVersions = NonEmptyList.of(Version("9.2.0")),
+      newerGroupId = Some(GroupId("com.github.jwt-scala")),
+      newerArtifactId = Some("jwt-play-json")
+    )
+    val duplicatedUpdates = Update.ForGroupId(NonEmptyList.of(artifactId, artifactId))
```

Comment:
What I don't know is, and also I did not investigate, is it expected / intended behaviour to generate duplicate updates? 

The dependency changes its group id and version 

`com.pauldijou` -> `com.github.jwt-scala`
`5.0.0` -> `9.2.0`

Reply:
No, it is not intended that we generate duplicated `Replacement`s. I think we should investigate why this is happening.

Reply:
I'll try to reproduce tomorrow... Hopefully I can reproduce ;)

Reply:
I found an `Update.ForArtifactId` (which Scala Steward could have produced) that reproduces the error reported here. I'll update the test case accordingly.

Reply:
By the whit this I could reproduce, https://github.com/Seetaramayya/scala-steward-bug-reproduction

```
2023-03-23 15:41:55,003 INFO  Using git version 2.38.4
2023-03-23 15:41:56,053 INFO  Using scalafix 0.10.4
2023-03-23 15:41:57,276 INFO  Using scalafmt 3.7.2
2023-03-23 15:41:58,207 INFO  Clean /opt/scala-steward/workspace/repos
2023-03-23 15:41:58,284 INFO  ──────────── Steward Seetaramayya/scala-steward-bug-reproduction ────────────
2023-03-23 15:41:58,286 INFO  Check cache of Seetaramayya/scala-steward-bug-reproduction
2023-03-23 15:41:59,406 INFO  Clone Seetaramayya/scala-steward-bug-reproduction
2023-03-23 15:42:00,798 INFO  Refresh cache of Seetaramayya/scala-steward-bug-reproduction
2023-03-23 15:42:00,867 INFO  Get dependencies in . from sbt
.
.
.

2023-03-23 18:21:07,975 INFO  Process update com.pauldijou:(jwt-play-json, jwt-play-json_2.12, jwt-play-json_2.13) : 5.0.0 -> 9.2.0
2023-03-23 18:21:08,890 INFO  Create branch update/jwt-play-json-9.2.0
(Replacement(Position(build.sbt,640,5.0.0),9.2.0),0)
(Replacement(Position(build.sbt,584,com.pauldijou),com.github.jwt-scala),1)
(Replacement(Position(build.sbt,612,jwt-play-json),jwt-play-json),2)
(Replacement(Position(build.sbt,584,com.pauldijou),com.github.jwt-scala),3)
(Replacement(Position(build.sbt,612,jwt-play-json),jwt-play-json),4)
2023-03-23 18:21:08,983 ERROR Steward Seetaramayya/scala-steward-bug-reproduction failed
java.lang.Throwable: failed to edit /opt/scala-steward/workspace/repos/Seetaramayya/scala-steward-bug-reproduction/build.sbt
	at org.scalasteward.core.io.FileAlg$$anonfun$editFile$4.applyOrElse(FileAlg.scala:69)
	at org.scalasteward.core.io.FileAlg$$anonfun$editFile$4.applyOrElse(FileAlg.scala:69)
	at scala.PartialFunction$AndThen.applyOrElse(PartialFunction.scala:288)
	at cats.ApplicativeError.$anonfun$recoverWith$1(ApplicativeError.scala:172)
	at flatMap @ org.scalasteward.core.io.FileAlg.editFile(FileAlg.scala:68)
	at recoverWith$extension @ org.scalasteward.core.io.process$.$anonfun$slurp$3(process.scala:82)
	at flatTraverse @ org.scalasteward.core.update.artifact.ArtifactMigrationsLoader.loadAll(ArtifactMigrationsLoader.scala:40)
	at flatTraverse @ org.scalasteward.core.update.artifact.ArtifactMigrationsLoader.loadAll(ArtifactMigrationsLoader.scala:40)
	at flatTraverse @ org.scalasteward.core.update.artifact.ArtifactMigrationsLoader.loadAll(ArtifactMigrationsLoader.scala:40)
	at flatMap @ org.scalasteward.core.edit.EditAlg.$anonfun$applyUpdateReplacements$3(EditAlg.scala:108)
	at map @ org.scalasteward.core.edit.EditAlg.applyUpdateReplacements(EditAlg.scala:106)
	at flatMap @ org.scalasteward.core.edit.EditAlg.applyUpdateReplacements(EditAlg.scala:106)
	at flatMap @ org.scalasteward.core.edit.EditAlg.$anonfun$applyUpdate$7(EditAlg.scala:64)
	at flatMap @ org.scalasteward.core.edit.EditAlg.$anonfun$applyUpdate$5(EditAlg.scala:61)
	at flatMap @ org.scalasteward.core.edit.EditAlg.$anonfun$applyUpdate$4(EditAlg.scala:58)
Caused by: java.lang.StringIndexOutOfBoundsException: begin 597, end 584, length 783
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:3319)
	at java.base/java.lang.String.substring(String.java:1874)
	at org.scalasteward.core.edit.update.data.Substring$Replacement$.$anonfun$applyAll$3(Substring.scala:49)
	at scala.collection.immutable.List.foreach(List.scala:333)
	at org.scalasteward.core.edit.update.data.Substring$Replacement$.applyAll(Substring.scala:48)
	at org.scalasteward.core.edit.EditAlg.$anonfun$applyUpdateReplacements$5(EditAlg.scala:109)
	at org.scalasteward.core.io.FileAlg.$anonfun$editFile$3(FileAlg.scala:68)
	at scala.Option.fold(Option.scala:263)
	at org.scalasteward.core.io.FileAlg.$anonfun$editFile$1(FileAlg.scala:68)
	... 11 common frames omitted
```

Reply:
Thanks for the logs!

> Process update com.pauldijou:(jwt-play-json, jwt-play-json_2.12, jwt-play-json_2.13) : 5.0.0 -> 9.2.0

This confirms my suspicion. The update is an `Update.ForArtifactId` with multiple `ArtifactId.crossName`s.